test/test_collections.py

```diff
@@ -339,5 +339,34 @@ def test_from_httplib_py2(self):
         self.assertEqual(d['www-authenticate'], 'asdf, bla')
         self.assertEqual(d.getlist('www-authenticate'), ['asdf', 'bla'])
 
+    def test_from_httplib_py2_invalid_first_header(self):
+        if six.PY3:
+            raise SkipTest("python3 has a different internal header implementation")
```

Comment:
Sure, but that's why I suggested rewriting the test to use the entire httplib stack: that way we can validate that we really do behave identically in both cases.

Are you comfortable writing that test @haikuginger?